batch/batch/server/database.py

```diff
@@ -123,3 +123,31 @@ def __init__(self, db):
 
     async def has_record(self, id):
         return await super().has_record({'id': id})
+
+    async def find_records(self, user, complete=None, success=None):
+        values = []
+        sql = "select batch.* from {self.name} as batch"
+        joins = []
+        wheres = []
+        if complete or success:
+            joins += "inner join {self._db.jobs.name} as job on batch.id == job.batch_id"
+            if complete is not None:
+                values += "Complete"
+                if complete:
+                    wheres += "job.state = %s"
+                else:
+                    wheres += "job.state != %s"
+            if success:
+                values += 0
```

Comment:
I don't think this code is correct. I think you need the logic from #6028 where the n_jobs, n_complete, n_failure, n_success is all explicitly tracked in the batch record using triggers.